src/check_config_provider/redis_config_provider.rs

```diff
@@ -88,6 +138,103 @@ impl RedisConfigProvider {
             }
         }
     }
+
+    async fn monitor_updates(
+        &self,
+        manager: Arc<Manager>,
+        partitions: &[RedisPartition],
+        shutdown: CancellationToken,
+    ) {
+        let mut conn = self
+            .redis
+            .get_multiplexed_tokio_connection()
+            .await
+            .expect("Unable to connect to Redis");
+        let mut interval = interval(self.check_interval);
+
+        while !shutdown.is_cancelled() {
+            let _ = interval.tick().await;
+
+            for partition in partitions.iter() {
+                let mut pipe = redis::pipe();
+                // We fetch all updates from the list and then delete the key. We do this
+                // atomically so that there isn't any chance of a race
+                let (config_upserts, config_deletes) = pipe
+                    .atomic()
+                    .hvals(&partition.update_key)
+                    .del(&partition.update_key)
+                    .query_async::<(Vec<Vec<u8>>, ())>(&mut conn)
+                    .await
+                    .unwrap_or_else(|err| {
+                        tracing::error!(?err, "redis_config_provider.redis_query_failed");
+                        (vec![], ())
+                    })
+                    .0 // Get just the LRANGE results
+                    .iter()
+                    .map(|payload| {
+                        rmp_serde::from_slice::<ConfigUpdate>(payload).map_err(|err| {
+                            tracing::error!(?err, "config_consumer.invalid_config_message");
+                            err
+                        })
+                    })
+                    .filter_map(Result::ok)
+                    .fold((vec![], vec![]), |(mut upserts, mut deletes), update| {
+                        match update.action {
+                            ConfigUpdateAction::Upsert => upserts.push(update),
+                            ConfigUpdateAction::Delete => deletes.push(update),
+                        }
+                        (upserts, deletes)
+                    });
```

Comment:
this is fine too. I actually didn't mind the partition as long as we had the match